vm.go

```diff
@@ -0,0 +1,60 @@
+/// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+const procPath = "/proc"
+
+var errFound = errors.New("found")
+
+var hypervisors = []string{"qemu"}
+
+// IsVMRunning looks in /proc for a hypervisor process that contains
+// the containerID in its command line
+func IsVMRunning(containerID string) bool {
+	err := filepath.Walk(procPath, func(path string, info os.FileInfo, err error) error {
+		if !info.IsDir() {
+			return filepath.SkipDir
+		}
+
+		cmdlineFile := filepath.Join(path, "cmdline")
+		if _, err := os.Stat(cmdlineFile); os.IsNotExist(err) {
+			return filepath.SkipDir
+		}
+
+		content, err := ioutil.ReadFile(cmdlineFile)
+		if err != nil {
+			return filepath.SkipDir
+		}
+
+		for _, hypervisor := range hypervisors {
+			if bytes.Contains(content, []byte(hypervisor)) &&
```

Comment:
I don't think this is going to be reliable since wouldn't this match my `grep` process if I ran `ps -ef|grep qemu`?

The problem is that you've got to find only those hypervisor processes that we care about (what if the system had `qemu-system_x86_64` and `qemu-lite-system_x86_64` processes running simultaneously?) As such, I wonder if this program should be told the full path to the hypervisor that is actually being used (which might be, say, `/usr/local/bin/qemu-lite-system_x86_64`).